### PR TITLE
Deprecate cylc 5 cycling syntax

### PR DESCRIPTION
**Pull request by benfitzpatrick**
We should add deprecation warnings for the nearly-2-years-old cylc 5 syntax.

### CODE REVIEW

**Comment by arjclark** [comment]
Looks OK to me. No problems from the test-battery in my environment. Over to you @hjoliver

**Comment by hjoliver** [comment]
Good.

**Comment by benfitzpatrick** [comment]
This closed #1907.

File: lib/cylc/cfgspec/suite.py [hunk 1]
```diff
@@ -475,6 +475,22 @@ def upg(cfg, descr):
...
         converter(
             lambda x: x, 'changed naming to reflect non-date-time cycling'))
+    u.deprecate(
+        '6.0.0',
+        ['scheduling', 'cycling']
+    )
+    u.deprecate(
+        '6.0.0',
+        ['scheduling', 'special tasks', 'sequential']
+    )
```
Comment (hjoliver) [inline lib/cylc/cfgspec/suite.py]: @benfitzpatrick - can you recall why you marked "sequential" as deprecated?  It isn't cylc-5 specific, and although I don't particularly like it, it is arguably useful - simpler than putting in many explicit previous-instance triggers - sometimes.

**Pull request merged**